multisig-wallet/project/contracts/multisig-contract/tests/utils/interface/core.rs

```diff
@@ -1,75 +1,69 @@
+use crate::utils::setup::{ContractCallParams, MultiSig, SignatureInfo, TransferParams, User};
 use fuels::{
-    accounts::wallet::WalletUnlocked,
-    prelude::ContractId,
-    programs::call_response::FuelCallResponse,
-    types::{Bits256, Identity},
+    accounts::wallet::WalletUnlocked, programs::call_response::FuelCallResponse, types::Identity,
 };
 
-use crate::utils::setup::{MultiSig, SignatureInfo, User};
-
 pub(crate) async fn constructor(
     contract: &MultiSig<WalletUnlocked>,
     users: Vec<User>,
 ) -> FuelCallResponse<()> {
     contract.methods().constructor(users).call().await.unwrap()
 }
 
-pub(crate) async fn _execute_transaction(
+pub(crate) async fn execute_transaction(
     contract: &MultiSig<WalletUnlocked>,
-    to: Identity,
-    value: u64,
-    data: Bits256,
-    signatures_data: Vec<SignatureInfo>,
-) -> FuelCallResponse<()> {
-    contract
+    contract_call_params: Option<ContractCallParams>,
+    signatures: Vec<SignatureInfo>,
+    target: Identity,
+    transfer_params: TransferParams,
+) -> Result<FuelCallResponse<()>, &'static str> {
```

Comment:
Why do we need error handling here?
I imagine there is a test where you check the `Err(string)` defined below appears in a panic msg. It's a string though and we shouldn't be tested for that in contracts so I imagine this is propagated from elsewhere. Either way it fails. 
I haven't seen the usage yet so it may be valid but my immediate impression is to remove since based on this code alone and how we usually test we wouldn't care to handle it

Reply:
Its because of the need to use `.set_contract_ids()` on a call, which only accepts `ContractId`s, however `target` is an `Identity`, as such I have to match on `target` and handle the case where `Identity` is not a `ContractId`; I chose to return an error here. The actual panicking test for making a call when `target` isn't a `ContractId` doesn't use this function

Reply:
My understanding is that this is purely used via our set up so we need to set things up properly or this error would catch that, is that correct?
If it is then it should be removed. We ought to ensure tests follow the correct branch flow.

Reply:
> My understanding is that this is purely used via our set up so we need to set things up properly or this error would catch that, is that correct?

The error would be triggered if `target` was set to an `Identity::Address` in a test when calling `execute_transaction()`, in which case the actual contract method can't be called because it requires the use of `set_contract_ids()`. As such when matching on `target`, I returned an error if it's an `Identity::Address`, as the call cannot be made.

Reply:
I suppose I could take `target` as a `ContractId` instead of `Identity` for this utility function instead.

Reply:
An alternative would be to just extract the underlying `Address` from an `Identity:Address` and convert it into a `contract_id`

Reply:
> An alternative would be to just extract the underlying `Address` from an `Identity:Address` and convert it into a `contract_id`

I've implemented this

Reply:
Chotto matte, so we're not supposed to set an `Address` and instead only use `ContractId`, is that correct? If so we shouldn't be hacking the address into an id and instead just `panic!("<error>")`